pandas/core/computation/align.py

```diff
@@ -62,7 +65,7 @@ def _any_pandas_objects(terms) -> bool:
     return any(isinstance(term.value, PandasObject) for term in terms)
 
 
-def _filter_special_cases(f):
+def _filter_special_cases(f) -> Callable[[F], F]:
```

Comment:
Looks like this is only used once to decorate the defined function below, so maybe can follow up by combining the two functions